benchmarks/dynamo/cachebench.py

```diff
@@ -0,0 +1,173 @@
+
+
+def get_compile_time(file: tempfile._TemporaryFileWrapper) -> float:
+    lines = file.readlines()
+    # Decode from byte string, remove new lines, parse csv
+    lines = [line.decode("utf-8").strip().split(",") for line in lines]
+    compilation_time_idx = lines[0].index("compilation_latency")
+    compilation_time = lines[1][compilation_time_idx]
+    return float(compilation_time)
+
+
+def _run_torchbench_from_args(model: str, args: list[str]) -> tuple[float, float]:
+    with fresh_inductor_cache():
+        env = os.environ.copy()
+        with tempfile.NamedTemporaryFile(suffix=".csv") as file:
+            args.append("--output=" + file.name)
+            logger.info(f"Performing cold-start run for {model}")
+            subprocess.check_call(args, timeout=TIMEOUT, env=env)
+            cold_compile_time = get_compile_time(file)
+
+        args.pop()
+        with tempfile.NamedTemporaryFile(suffix=".csv") as file:
+            args.append("--output=" + file.name)
+            logger.info(f"Performing warm-start run for {model}")
+            subprocess.check_call(args, timeout=TIMEOUT, env=env)
+            warm_compile_time = get_compile_time(file)
+
+        return cold_compile_time, warm_compile_time
+
+
+def _run_torchbench_model(results: list[RunResult], model: str) -> None:
+    cur_file = os.path.abspath(__file__)
+    torchbench_file = os.path.join(os.path.dirname(cur_file), "torchbench.py")
+    assert os.path.exists(
+        torchbench_file
+    ), f"Torchbench does not exist at {torchbench_file}"
+
+    base_args = [
+        sys.executable,
+        torchbench_file,
+        f"--only={model}",
+        "--repeat=5",
+        "--performance",
+        "--backend=inductor",
+    ]
+    for mode in ["inference", "training"]:
+        for dynamic in [False, True]:
+            args = list(base_args)
+            if mode == "inference":
+                args.extend(["--inference", "--bfloat16"])
+            else:
+                assert mode == "training"
+                args.extend(["--training", "--amp"])
+            if dynamic:
+                args.extend(["--dynamic-shapes", "--dynamic-batch-only"])
+
+            logger.info(f"Command: {args}")
+            try:
+                cold_compile_t, warm_compile_t = _run_torchbench_from_args(model, args)
+                results.append(
+                    RunResult(
+                        "model",
+                        mode,
+                        dynamic,
+                        cold_compile_t,
+                        warm_compile_t,
+                        cold_compile_t / warm_compile_t,
+                    )
+                )
+            except Exception as e:
+                print(e)
+                return None
+
+
+def _write_results_to_json(results: list[RunResult], output_filename: str) -> None:
+    records = []
+    for result in results:
+        for metric_name in ["cold_compile_time(s)", "warm_compile_time(s)", "speedup"]:
```

Comment:
nit: These if statements seem kinda silly, just do for (metric_name, value) in [("cold_compile_time", result.cold_compile_s), ()] 
etc
You can use a constant dict instead of a list of tuples too